clang/lib/3C/3C.cpp

```diff
@@ -522,6 +522,15 @@ bool _3CInterface::addVariables() {
 
   std::lock_guard<std::mutex> Lock(InterfaceMutex);
 
+  // Find multi-decls and assign names to unnamed inline TagDecls now so that
+  // the assigned type names are available when we construct ConstraintVariables
+  // for the multi-decl members in the "Add Variables" step below.
+  // REVIEW: Move to a separate _3CInterface pass?
+  for (auto &TU : ASTs)
+    GlobalProgramInfo.TheMultiDeclsInfo.findUsedTagNames(TU->getASTContext());
+  for (auto &TU : ASTs)
+    GlobalProgramInfo.TheMultiDeclsInfo.findMultiDecls(TU->getASTContext());
```

Comment:
Good point.  In this case, I knew that `findUsedTagNames` and `findMultiDecls` could never fail (except via assertions that terminate 3C immediately), but as you suggest, this won't be obvious to future readers.  I was hoping to avoid the boilerplate of making each of these a separate `_3CInterface` method call, which would require extra code all the way to `3CStandalone.cpp`, though I'm willing to do that if we think it's important to have separate "Failure occurred while X" messages for these steps (even though they would never be reached in the current code).

I realize a good compromise may be to add intermediate `if (!isSuccessfulSoFar()) return false;` checks, as we have in several other `_3CInterface` passes.  I've gone ahead and done this.  Are you satisfied with that approach?